File: Intermediate/Topic-004-Arrays-Prefix-Sum/Assigment/Q1-Range-Sum-Query/Range-Sum-Query.c
/**
 * @input A : Integer array
 * @input n1 : Integer array's ( A ) length
 * @input B : 2D integer array 
 * @input n21 : Integer array's ( B ) rows
 * @input n22 : Integer array's ( B ) columns
 * 
 * @Output Long Array. You need to malloc memory, and fill the length in len1
 */
long long* rangeSum(int* A, int n1, int** B, int n21, int n22, int *len1) {

    // Create a prefix sum array

    long long *prefix = (long long *) malloc ( n1 * sizeof( long long));

    // update the prefix sum array

    prefix[0] = A[0];
    int i;
    for (i = 1; i < n1; i++) {
        prefix[i] = prefix[i - 1] + A[i];
    }

       // Create a result array

    long long *result = (long long *) malloc(n21 * sizeof(long long));

    //    answer the query
    for (i = 0; i < n21; i++) {
        int l = B[i][0];
        int r = B[i][1];

        if ( l == 0 ) {
            result[i] = prefix[r];
        } else {
            result[i] = prefix[r] - prefix[l - 1];
        }
    }
    //  Set length
    *len1 = n21;
    return result;
}

File: Intermediate/Topic-004-Arrays-Prefix-Sum/Assigment/Q1-Range-Sum-Query/Range-Sum-Query.py
class Solution:
    # @param A : list of integers
    # @param B : list of list of integers
     # @return an list of long
    def rangeSum(self, A, B):
        # Create an array p to store the prefix sum array
        p = [0] * len(A)

        p[0] = A[0]

        # create prefix sum array
        for i in range(1, len(A)):
            p[i] = p[i -1] + A [i]

        
        # Query the Answer
        sum = [0] * len(B)
        for i in range(len(B)):
            l = B[i][0] 
            r = B[i][1]

            if  l == 0 :
                sum[i] = p[r]
            else :
                 sum [i] = p[r] - p[l -1]
        return sum

